release/models/relay-agent/openconfig-relay-agent.yang

```diff
@@ -624,6 +633,148 @@ module openconfig-relay-agent {
 
   }
 
+  grouping relay-agent-ipv4-helper-address-config {
+    description
+      "Configuration options for an IP helper-address";
+
+    leaf address {
+      type oc-inet:host;
+      description
+        "IP address to relay DHCP requests to";
+    }
+
+    leaf source-address {
+      type union {
+        type inet:ipv4-address;
+        type oc-if:interface-id;
+      }
+      description
+        "IP address or interface whose IP address is to be used as source in
+	relayed DHCP requests";
+    }
+
+    leaf network-instance {
+      type oc-ni:network-instance-ref;
+      description
+        "The network instance that this DHCP Server
+        is associated with";
+    }
+  }
+
+  grouping relay-agent-ipv6-helper-address-config {
+    description
+      "Configuration options for an IPv6 helper-address";
+
+    leaf address {
+      type oc-inet:host;
+      description
+        "IPv6 address to relay DHCPv6 requests to";
```

Comment:
The use of `oc-inet:host` implies a union of `domain-name` as well as `ipv4-address` _AND_ `ipv6-address`

Reply:
This should be oc-inet:ip-address and not host

Reply:
How many implementations support domain-name for such use (that also store the object as such to reflect in config/state and not resolve at time of input)?  And is it a good idea to rely off DNS (and/or a local cache) in such operational scenarios?

IMO precise IP address types should suffice here.  Speaking on behalf of one implementation, we do not support DNS names for this purpose and baking that into a union is not deviate-able without replacing the entire type (and thus changing the underlying namespace for alternate NBI where this matters) - it is then left to runtime error handling which is also not conveyed at the data modeling layers